apps/prairielearn/src/pages/instructorCourseAdminInstances/instructorCourseAdminInstances.html.ts

```diff
@@ -121,6 +121,28 @@ export function InstructorCourseAdminInstances({
                 </tbody>
               </table>
             </div>
+            ${courseInstances.length === 0
+              ? html`
+                  <div class="my-4">
+                    <p class="text-center font-weight-bold">No course instances found.</p>
+                    <p class="text-center mb-0">
+                      A course instance is a single offering of a course, such as "Fall 2024" or
+                      "Fall 2024, Section M".
+                    </p>
+                    <p class="text-center">
+                      To create one, click <strong>Add course instance</strong>.
+                    </p>
```

Comment:
The "Add course button" is actually conditionally rendered, see lines 35-37:

- It isn't shown for the example course.
- It isn't shown if the user doesn't have Editor permissions for the course.
- It isn't shown if the course path doesn't exist on disk (this is the `resLocals.needToSync` flag).

We often find that people get confused about why they can't do something, so it'd improve the UX to show specific messages in these cases. For instance, if `resLocals.authz_data.has_course_permission_edit` is `false`, we can say something like "Course Editors can create new course instances". How does that sound?

Reply:
The UI now displays messages in the cases where the Add course instance button is hidden:
- Doesn't have Editor permissions: "Course Editors can create new course instances."
- Example course: "You can't add course instances to the Example Course."
- Needs to Sync: "Sync the course from the Sync page before creating a new course instance."

Let me know if I should adjust any of these messages.

Reply:
Looking pretty good! I'll leave some comments on the specific messages; we can continue conversation there.